src/main/java/com/mraof/minestuck/tileentity/ProspitTransportalizerTileEntity.java

```diff
@@ -0,0 +1,91 @@
+package com.mraof.minestuck.tileentity;
+
+import com.mojang.datafixers.Dynamic;
+import com.mraof.minestuck.block.MSBlocks;
+import com.mraof.minestuck.util.Teleport;
+import com.mraof.minestuck.world.MSDimensions;
+import net.minecraft.block.Block;
+import net.minecraft.entity.Entity;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.nbt.NBTDynamicOps;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.GlobalPos;
+import net.minecraft.world.World;
+import net.minecraft.world.dimension.DimensionType;
+import net.minecraft.world.gen.Heightmap;
+import net.minecraft.world.server.ServerWorld;
+import net.minecraftforge.common.DimensionManager;
+
+public class ProspitTransportalizerTileEntity extends TileEntity //implements ITeleporter
```

Comment:
So I guess the big question here is, why would the prospit/derse transportalizers work like the skaia portal does, rather than using the existing transportalizer block/tile entity? What is the end goal here?
On a smaller note, even if you do continue having the prospit/derse transportalizers working distinct from standard transportalizers, since the prospit and serse variants are so similar, I'd want the bulk of the prospit and derse transportalizers to be implemented under one class, rather than two. (DRY coding principle)